src/Vector.cpp

```diff
@@ -0,0 +1,207 @@
+#include "Vector.h"
+
+
+// Constructor
+Vector::Vector(size_t size, int value)
+    : m_size(size)
+    , m_capacity(get_capacity(size))
+    , m_data(new int[m_capacity])
+{
+    // Initialize all elements to the given value
+    for (size_t i = 0; i < m_size; ++i) {
+        m_data[i] = value;
+    }
+}
+
+// Copy constructor
+Vector::Vector(const Vector& other)
+    : m_size(other.m_size)
+    , m_capacity(other.m_capacity)
+    , m_data(new int[m_capacity])
+{
+    // Copy all elements from the other vector
+    for (size_t i = 0; i < m_size; ++i) {
+        m_data[i] = other.m_data[i];
+    }
+}
+
+// Move constructor
+Vector::Vector(Vector&& other) noexcept
+    : m_size(other.m_size)
+    , m_capacity(other.m_capacity)
+    , m_data(other.m_data)
+{
+    // Take ownership of the other vector's data
+    other.m_size = 0;
+    other.m_capacity = 0;
+    other.m_data = nullptr;
+}
+
+// Copy assignment operator
+Vector& Vector::operator=(const Vector& other) {
+    if (this != &other) {
+        // Copy and swap idiom for exception safety
+        Vector temp(other);
+        std::swap(m_size, temp.m_size);
+        std::swap(m_capacity, temp.m_capacity);
+        std::swap(m_data, temp.m_data);
+    }
+    return *this;
+}
+
+// Move assignment operator
+Vector& Vector::operator=(Vector&& other) noexcept {
+    if (this != &other) {
+        // Free the old data and take ownership of the new data
+        delete[] m_data;
+        m_size = other.m_size;
+        m_capacity = other.m_capacity;
+        m_data = other.m_data;
+        other.m_size = 0;
+        other.m_capacity = 0;
+        other.m_data = nullptr;
+    }
+    return *this;
+}
+
+// append an element to the end of the vector
+void Vector::push_back(int value) {
+    if (m_size == m_capacity) {
+        if (m_capacity < 128) {
+            reserve(m_capacity * 2);
+        }
+        else {
+            reserve(m_capacity * 3 / 2);
+        }
+    }
+    m_data[m_size++] = value;
+}
+
+// remove the last element from the vector
+void Vector::pop_back() {
+    if (!empty()) {
+        // Remove the last element from the vector and update the size
```

Comment:
Did this remove the element? What if now the user tries accessing the removed element?
I.e: 
```cpp
Vector v{1,2};
v.pop_back();
std::cout << v[1]; // What will happen here? 
```